pkg/middleware/http/ratelimiter.go

```diff
@@ -181,21 +191,38 @@ func (t *apiLimiter) HTTPAllow(ctx context.Context, r *http.Request) bool {
 func Limit(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if !limiter.Allow(context.Background(), r) {
-			w.Header().Set("Content-Type", "application/json")
-			w.WriteHeader(http.StatusTooManyRequests)
+			MakeLimitErrorResponse(w, ErrTooManyRequest)
 			return
 		}
-
 		if !limiter.HTTPAllow(context.Background(), r) {
-			w.Header().Set("Content-Type", "application/json")
-			w.WriteHeader(http.StatusTooManyRequests)
+			MakeLimitErrorResponse(w, ErrTooManyRequest)
 			return
 		}
-
 		next.ServeHTTP(w, r)
 	})
 }
 
+func MakeLimitErrorResponse(w http.ResponseWriter, err error) {
+	gfspErr := gfsperrors.MakeGfSpError(err)
+	var xmlInfo = struct {
+		XMLName xml.Name `xml:"Error"`
+		Code    int32    `xml:"Code"`
+		Message string   `xml:"Message"`
+	}{
+		Code:    gfspErr.GetInnerCode(),
+		Message: gfspErr.GetDescription(),
+	}
+	xmlBody, err := xml.Marshal(&xmlInfo)
+	if err != nil {
+		log.Errorw("failed to marshal error response", "error", gfspErr.String())
```

Comment:
Refine to `"gfsp_error", gfspErr.String(), "error", err`. It is best for the log to include context and specific error reasons.